doc/api/os.md

```diff
@@ -508,6 +508,28 @@ On POSIX systems, the operating system release is determined by calling
 available, `GetVersionExW()` will be used. See
 <https://en.wikipedia.org/wiki/Uname#Examples> for more information.
 
+## `os.guessFileDescriptorType(fd)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fd` {integer} The file descriptor number to try and guess the type of.
+
+* Returns: {string|null}
+
+Returns the type of the file descriptor passed in, or `null` if the provided file descriptor
+is invalid.
```

Comment:
Nit: some expanded explanation about why this is useful would probably be helpful. It's rather obscure.